### ISSUE TITLE
Universal link `apple-app-site-association` misses `apps` property

### ISSUE THREAD

**Issue by Chralu** [flutter/website#8636]
### Page URL

https://docs.flutter.dev/cookbook/navigation/set-up-universal-links#apple-app-site-association

### Describe the problem

To get Universal links working on iOS, you need to add an `apps` property to the `apple-app-site-association` file.

So, the fils should look like this : 

```json
{
  "applinks": {
      "apps": [],
      "details": [
      {
        "appID": "S8QB4VV633.com.example.deeplinkCookbook",
        "paths": ["*"]
      }
    ]
  }
}
```

### Expected fix

Maybe the website doc should be updated.

**Comment by danagbemava-nc** [comment]
This might be the old syntax as I see something different in the current [docs](https://developer.apple.com/documentation/xcode/supporting-associated-domains). I do see the proposed syntax in the documentation [archive](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html)

**Comment by Chralu** [comment]
@danagbemava-nc You're right, I checked the documentation archive without noticing...

Nevertheless, I noticed that without that `apps` property, universal links does not work on iOS.
It needs to be further tested...

**Comment by Chralu** [comment]
[This conversation](https://stackoverflow.com/questions/68033065/how-to-format-and-serve-an-apple-app-site-association-file-correctly) is interesting.

TL;DR : There are indeed two `apple-app-site-association` formats (from **archive doc**, and **current doc**). 
And [the Flutter doc](https://docs.flutter.dev/cookbook/navigation/set-up-universal-links#apple-app-site-association) shows a mix of both (which doesn't seem to work)




### [Archive Documentation format](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html#//apple_ref/doc/uid/TP40016308-CH12-SW1)
```json
{
    "applinks": {
        "apps": [],
        "details": [
            {
                "appID": "9JA89QQLNQ.com.apple.wwdc",
                "paths": [ "/wwdc/news/", "/videos/wwdc/2015/*"]
            },
            {
                "appID": "ABCD1234.com.apple.wwdc",
                "paths": [ "*" ]
            }
        ]
    }
}
```

### [Current Documentation format](https://developer.apple.com/documentation/xcode/supporting-associated-domains)

```json
{
  "applinks": {
      "details": [
           {
             "appIDs": [ "ABCDE12345.com.example.app", "ABCDE12345.com.example.app2" ],
             "components": [
               {
                  "#": "no_universal_links",
                  "exclude": true,
                  "comment": "Matches any URL whose fragment equals no_universal_links and instructs the system not to open it as a universal link"
               },
               {
                  "/": "/buy/*",
                  "comment": "Matches any URL whose path starts with /buy/"
               },
               {
                  "/": "/help/website/*",
                  "exclude": true,
                  "comment": "Matches any URL whose path starts with /help/website/ and instructs the system not to open it as a universal link"
               },
               {
                  "/": "/help/*",
                  "?": { "articleNumber": "????" },
                  "comment": "Matches any URL whose path starts with /help/ and which has a query item with name 'articleNumber' and a value of exactly 4 characters"
               }
             ]
           }
       ]
   },
   "webcredentials": {
      "apps": [ "ABCDE12345.com.example.app" ]
   },

    "appclips": {
        "apps": ["ABCED12345.com.example.MyApp.Clip"]
    }
}
```
